### PR TITLE
Feature/norwegian contracts

### PR DESCRIPTION
**Pull request by meletios**
Adds new contracts
- Home Content ✅
- Travel ✅
- add some tests for creating contracts 🚧

### CODE REVIEW

File: src/main/java/com/hedvig/contract/service/ContractServiceImpl.kt [hunk 1]
```diff
@@ -6,33 +6,44 @@ import com.hedvig.contract.shared.models.AgreementState
...
     return contractStateRepository.findByHolderMemberId(holderMemberId)
   }
 
+  override fun getCurrentOrLastAgreement(holderMemberId: String): AgreementState {
+    return contractStateRepository.findByHolderMemberId(holderMemberId).first().pendingOrCurrentOrNextOrLastAgreement
+  }
+
+  override fun getContractsByHolderMemberId(holderMemberId: String): List<Contract> {
+    val contractStates = getContractStatesByHolderMemberId(holderMemberId)
+    return contractStates.map { contractState -> typeMapper.toContract(contractState) }
```
Comment (johantjellden) [inline src/main/java/com/hedvig/contract/service/ContractServiceImpl.kt]: For this one i would maybe return the ContractState and do the mapping in the controller. Even tough we usually do this it is a bit weird that ConstractService returns `...web.dtos.Contract`

File: src/main/java/com/hedvig/productPricing/service/query/ProductRepository.java [hunk 2]
```diff
@@ -15,6 +16,7 @@
 import java.util.UUID;
 
 @Transactional
+@Repository
```
[the remark this answers is not in the document]
Comment (johantjellden) [reply src/main/java/com/hedvig/productPricing/service/query/ProductRepository.java]: If you annotate it with @Repository spring will catch persistence exceptions and re-throw them as  some "unified spring exceptions".
https://www.baeldung.com/spring-component-repository-service

File: src/main/java/com/hedvig/contract/service/ContractServiceImpl.kt (hunk 1 above)
Comment (vonElfvin) [reply src/main/java/com/hedvig/contract/service/ContractServiceImpl.kt]: Moved the mapping to the controller

File: src/main/java/com/hedvig/contract/service/PremiumService.kt [hunk 3]
```diff
@@ -0,0 +1,8 @@
+package com.hedvig.contract.service
+
+import com.hedvig.contract.web.dtos.AgreementPremiumCost
+import java.time.YearMonth
+
+interface PremiumService {
```
Comment (FredrikAreschoug) [inline src/main/java/com/hedvig/contract/service/PremiumService.kt]: :sparkler:

File: src/main/java/com/hedvig/contract/configuration/AxonConfig.kt [hunk 4]
```diff
@@ -34,4 +42,31 @@ class AxonConfig {
...
+    xStreamSerializer.addAlias(
+      "com.hedvig.contract.apartments.models.ApartmentLineOfBusiness",
+      SwedishApartmentLineOfBusiness::class.java
+    )
+    xStreamSerializer.addAlias(
+      "com.hedvig.contract.houses.aggregates.HouseAgreement",
+      SwedishHouseAgreement::class.java
+    )
+    xStreamSerializer.addAlias(
+      "com.hedvig.contract.util.SignSource>",
```
Comment (FredrikAreschoug) [inline src/main/java/com/hedvig/contract/configuration/AxonConfig.kt]:
```suggestion
      "com.hedvig.contract.util.SignSource",
```

File: src/main/java/com/hedvig/contract/service/AgreementServiceImpl.kt [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+import com.hedvig.contract.web.dtos.agreements.AddAgreementRequest
+import com.hedvig.contract.web.dtos.agreements.ChangeFromDateOnAgreementRequest
+import com.hedvig.contract.web.dtos.agreements.ChangeToDateOnAgreementRequest
+import com.hedvig.contract.web.dtos.agreements.ReplaceAgreementRequest
+import org.axonframework.commandhandling.gateway.CommandGateway
+import org.springframework.stereotype.Service
+import java.util.UUID
+
+@Service
+class AgreementServiceImpl(
```
Comment (FredrikAreschoug) [inline src/main/java/com/hedvig/contract/service/AgreementServiceImpl.kt]: Should we test this one?

File: src/main/java/com/hedvig/contract/service/PremiumServiceImpl.kt [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+import com.hedvig.contract.web.dtos.AgreementPremiumCost
+import org.javamoney.moneta.Money
+import org.springframework.stereotype.Service
+import java.math.BigDecimal
+import java.math.RoundingMode
+import java.time.YearMonth
+import javax.money.MonetaryAmount
+
+@Service
+class PremiumServiceImpl(
```
Comment (FredrikAreschoug) [inline src/main/java/com/hedvig/contract/service/PremiumServiceImpl.kt]: Same here should we have tests?

File: src/main/resources/schema.graphqls [hunk 7]
```diff
@@ -100,3 +332,5 @@ type MonetaryAmountV2 {
 
 # A String-representation of `java.time.LocalDate`, ex:  `"2018-09-26"`
 scalar LocalDate
+# A Unix timestamp-representation of `java.time.Instant`. ex: `"1582894553"`
+scalar Instant
```
Comment (FredrikAreschoug) [inline src/main/resources/schema.graphqls]: Dosen't this one collide with graphql-commons scalar of instant?

File: src/main/java/com/hedvig/contract/web/dtos/contracts/CreateContractRequest.kt [hunk 8]
```diff
@@ -0,0 +1,17 @@
...
+  val memberId: String,
+  val firstName: String,
+  val lastName: String,
+  val ssn: String,
+  val currentInsurer: String?,
+  val signSource: SignSource?,
+  val referenceToken: String?,
+  val signature: String?,
+  val oscpResponse: String?,
+  val quotes: List<AgreementQuote>
```
Comment (FredrikAreschoug) [inline src/main/java/com/hedvig/contract/web/dtos/contracts/CreateContractRequest.kt]: Nice! but maybe rename it to `CreateContractsRequest`

File: src/main/java/com/hedvig/contract/service/ContractServiceImpl.kt [hunk 9]
```diff
@@ -1,43 +1,109 @@
...
     return agreementStateRepository.findByIdOrNull(agreementId)?.contract
       ?: throw ContractNotFoundException("No contract found with agreementId $agreementId")
   }
 
-  override fun getContractsByHolderMemberId(holderMemberId: String): List<ContractState<AgreementState>> {
+  override fun getContractStatesByHolderMemberId(holderMemberId: String): List<ContractState<AgreementState>> {
     return contractStateRepository.findByHolderMemberId(holderMemberId)
   }
 
+  override fun createContractsMaybe(request: CreateContractRequest): List<CreateContractResponse> {
```
Comment (FredrikAreschoug) [inline src/main/java/com/hedvig/contract/service/ContractServiceImpl.kt]: Test for this one <3

File: src/main/java/com/hedvig/contract/service/AgreementServiceImpl.kt (hunk 5 above)
Comment (vonElfvin) [reply src/main/java/com/hedvig/contract/service/AgreementServiceImpl.kt]: Given that all methods only contain one method call and thus will be mocked in a test I don't see the gain in that. Would love to hear your opinion on that however.

File: src/main/java/com/hedvig/contract/service/PremiumServiceImpl.kt (hunk 6 above)
Comment (vonElfvin) [reply src/main/java/com/hedvig/contract/service/PremiumServiceImpl.kt]: This one should really be tested yes

Comment (vonElfvin) [reply src/main/java/com/hedvig/contract/service/PremiumServiceImpl.kt]: Added tests now

File: src/main/resources/schema.graphqls (hunk 7 above)
Comment (vonElfvin) [reply src/main/resources/schema.graphqls]: I'm not sure? It seems to work? 👀

File: src/main/java/com/hedvig/contract/service/ContractServiceImpl.kt (hunk 9 above)
Comment (vonElfvin) [reply src/main/java/com/hedvig/contract/service/ContractServiceImpl.kt]: Added some tests

**Pull request merged**